lib/utils.js

```diff
@@ -48,10 +50,26 @@ try {
   console.warn(err.message);
 }
 
+/*
+ * Checks that collation is supported by server.
+ *
+ * @param {Server} [server] to check against
+ * @param {object} [cmd] object where collation may be specified
+ * @param {function} [callback] callback function
+ * @return true if server does not support collation
+ */
+const collationNotSupported = (server, cmd, callback) => {
+  if (cmd && cmd.collation && server.ismaster && server.ismaster.maxWireVersion < 5) {
+    callback(new MongoError(f('server %s does not support collation', server.name)));
+    return true;
```

Comment:
I can't thing of a better solution, but I'd like to have a discussion about invoking the callback and returning a boolean to help the invokee return early. CC @mbroadst @kvwalker

Reply:
I'm conflicted as well. Removing returning the `MongoError` at the actual call site makes it feel like we are decreasing readability: on reading it's not immediately clear that `callback` is ever called, though it is passed somewhere.

Reply:
Alternatively, could we just have the `collationNotSupported` function just return the boolean indicating whether collation is supported? Then the code might look like: 
`if collationNotSupported(server, cmd) return callback (new MongoError('...'));`
Although I guess that still leaves repetitive code

Reply:
I think my preference is to leave the callbacks in place, and reduce this function to determining if collation is supported or not. It gives us more control over what message is returned, and whether we want to throw or return an async error. I also think it improves readability. I'm about a 4/10 on this, but let's see what y'all think.